search-commons/src/main/java/no/unit/nva/search2/common/QueryFilter.java

```diff
@@ -4,28 +4,33 @@
 import org.opensearch.index.query.QueryBuilder;
 import org.opensearch.index.query.QueryBuilders;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
 
 public class QueryFilter {
-    private final transient List<QueryBuilder> filters = new ArrayList<>();
+    private final transient Map<String, QueryBuilder> filters = new HashMap<>();
 
     public QueryFilter() {
     }
 
     public BoolQueryBuilder get() {
         var boolQueryBuilder = QueryBuilders.boolQuery();
-        filters.forEach(boolQueryBuilder::must);
+        filters.values().forEach(boolQueryBuilder::must);
         return boolQueryBuilder;
     }
 
     public void set(QueryBuilder... filters) {
         this.filters.clear();
-        this.filters.addAll(List.of(filters));
+        var filterMap = Arrays.stream(filters)
+            .map(filter -> Map.entry(filter.queryName(), filter))
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        this.filters.putAll(filterMap);
```

Comment:
perhaps something like

        this.filters.clear();
        Arrays.stream(filters)
            .forEach(this::add)

Reply:
maybe jsut .foreach even? no stream

Reply:
One would think that was a natural function for an array, but in Java, **no** 🤷